packages/apollo-federation/src/composition/compose.ts

```diff
@@ -464,13 +471,28 @@ export function addFederationMetadataToSchemaNodes({
       },
     };
   }
+
+  // add all definitions of a specific directive for validation later
+  for (const directiveName of Object.keys(directiveDefinitionsMap)) {
+    const directive = schema.getDirective(directiveName);
+    if (!directive) continue;
+
+    directive.federation = {
```

Comment:
This can be simplified! Particularly the nested bits are overspread for sure.

Nothing else is using the federation key for now, but would you still spread the top level for future additions? More of a theoretical question than anything 😄

```js
directive.federation = {
  // ...directive.federation,
  directiveDefinitions: directiveDefinitionsMap[directiveName],
};
```